components/raftstore/src/store/worker/split_check.rs

```diff
@@ -335,6 +348,28 @@ where
             Task::ChangeConfig(c) => self.change_cfg(c),
             #[cfg(any(test, feature = "testexport"))]
             Task::Validate(f) => f(&self.cfg),
+            Task::GetRegionApproximateSizeAndKeys {
+                region,
+                pending_tasks,
+                cb,
+            } => {
+                if pending_tasks.fetch_sub(1, AtomicOrdering::SeqCst) > 1 {
+                    return;
+                }
+                let size =
+                    get_region_approximate_size(&self.engine, &region, 0).unwrap_or_default();
+                let keys =
+                    get_region_approximate_keys(&self.engine, &region, 0).unwrap_or_default();
+                let _ = self.router.send(
+                    region.get_id(),
+                    CasualMessage::RegionApproximateSize { size },
+                );
+                let _ = self.router.send(
+                    region.get_id(),
+                    CasualMessage::RegionApproximateKeys { keys },
+                );
```

Comment:
These data may be sent to a different peer instead of the expected one in rare circumstances. I find it is the same as the original code of `RegionApproximateSize/Keys`, which also has this problem. It seems that the side effect is little but it's risky. I will file an issue for that.

Reply:
Sounds like a potential risk. Please let me know what caused it to happen when you are free.